boefjes/boefjes/plugins/kat_dns_version/main.py

```diff
@@ -0,0 +1,49 @@
+"""Boefje script for getting namserver version"""
+
+import json
+import logging
+from os import getenv
+
+import dns
+
+from boefjes.job_models import BoefjeMeta
+
+logger = logging.getLogger(__name__)
+
+
+def run(boefje_meta: BoefjeMeta) -> list[tuple[set, str | bytes]]:
+    input_ = boefje_meta.arguments["input"]  # input is IPService
+    ip_port = input_["ip_port"]
+    if input_["service"]["name"] != "dns":
+        logging.info("Not a DNS service")
+        raise StopIteration
+
+    ip = ip_port["address"]["address"]
+    port = ip_port["port"]
+    protocol = ip_port["protocol"]
+
+    timeout = getenv("TIMEOUT", 30)
```

Comment:
The inferred type of this variable is now `str | int`, whilst the function below expects a float. I suggest implementing this:
```suggestion
    timeout = float(getenv("TIMEOUT", 30))
```